kuma/javascript/src/newsletter.jsx

```diff
@@ -221,15 +235,14 @@ export default function Newsletter() {
                             ref={errorsRef}
                             id="newsletter-errors"
                             className={
-                                errorsArray ? 'newsletter-errors' : 'hidden'
+                                hasErrors ? 'newsletter-errors' : 'hidden'
                             }
-                            aria-hidden={errorsArray ? true : false}
+                            aria-hidden={hasErrors}
                         >
                             <ul className="errorlist">
-                                {errorsArray &&
-                                    errorsArray.map((error, index) => (
-                                        <li key={index}>{error}</li>
-                                    ))}
+                                {errors.map((error, index) => (
+                                    <li key={index}>{error}</li>
```

Comment:
Don't use indexes for keys. If the first render, the `errors` are:
```
['Foo', 'Bar']
```
It will produce:
```html
<li>Foo</li>
<li>Bar</li>
```
and then a form submission happens and now `errors` is: 
```
['One', 'Two', 'Three']
```
It will produce (in the second render):

```html
<li>Foo</li>
<li>Bar</li>
<li>Three</li>
```

Reply:
I don't think it would. Keys don't prevent reconciliation (i.e. finding out what changed) but rather make sure React doesn't lose track of which component is which. But this is indeed a potential source of future bugs if it's ever not a simple `li` element anymore but rather something with state (which could then lead to a state mixup). I'll change the keys to `index + error` which I think makes the most sense here.

Reply:
Threw together a fiddle, hope it has more explanatory than confusatory power:
https://codesandbox.io/s/naughty-banach-txlxc